python/cudf_polars/cudf_polars/callback.py

```diff
@@ -47,10 +63,26 @@ def default_memory_resource(device: int) -> rmm.mr.DeviceMemoryResource:
     -------
     rmm.mr.DeviceMemoryResource
         The default memory resource that cudf-polars uses. Currently
-        an async pool resource.
+        a managed memory resource, if `POLARS_GPU_ENABLE_CUDA_MANAGED_MEMORY`
+        environment variable is set `0`, then an async pool resource is returned.
+    bool
+        A flag indicating whether to enable prefetching.
     """
     try:
-        return rmm.mr.CudaAsyncMemoryResource()
+        if (
+            _env_get_int("POLARS_GPU_ENABLE_CUDA_MANAGED_MEMORY", default=1) == 1
+            and pylibcudf.utils._is_concurrent_managed_access_supported()
+        ):
+            free_memory, _ = rmm.mr.available_device_memory()
+            free_memory = int(round(float(free_memory) * 0.80 / 256) * 256)
```

Comment:
Can we document the reason for this math in the docstring so people know what pool size is being selected since we're implicitly pooling with 80% of the size by default.

Reply:
Done